irods_specific_implementation/irods_rest_api_base.h

```diff
@@ -132,6 +132,11 @@ namespace irods::rest
             load_client_api_plugins();
         } // ctor
 
+        bool to_bool(const std::string& s) noexcept
+        {
+            return s == "1" || s == "true";
+        }
+
```

Comment:
This feels out of place.

Consider moving it to utils.hpp.

The use of `std::string` could be replaced with `std::string_view` to avoid potential memory allocations.

Reply:
Interesting. If we pass ```s``` by reference, why would there be additional allocation?

Reply:
Added to the base class because adding to utils.hpp led to symbol clash with the include in ```...ApiImpl``` -- currently tinkering with CMake with the goal of pulling utils into separate .hpp and .cpp files.

Reply:
> Interesting. If we pass `s` by reference, why would there be additional allocation?

`std::string` will result in memory allocation if/when the SSO cannot be applied. For example, the clang compiler may store short strings (<= 16 bytes) on the stack. This makes short string use very fast. However, if a string is passed and exceeds that limit, we have to allocate memory on the heap.

`std::string_view` does not have this issue because it is designed as a view into the string or string literal. It never allocates memory. It simply wraps the string and exposes member functions that never modify the underlying string. `std::string_view` is also lightweight, making the copying of it acceptable.

> Added to the base class because adding to utils.hpp led to symbol clash with the include in ...ApiImpl -- currently tinkering with CMake with the goal of pulling utils into separate .hpp and .cpp files.

Which symbols collide?

Reply:
> Which symbols collide?

I should say instead that it's a double-include: in both ApiImpl and the "specific implementation" header. I then pulled apart ```utils``` into header and source files, and added utils.cpp to the source files for the executable. But I get an error saying that clang can't find a definition for, ```handle_request```. Perhaps I want to turn ```utils``` into a small library and link against that?

Reply:
The most recently pushed version represents my latest attempt to get CMake to do my bidding. Currently, I get the following error which I'm not sure how to resolve when running ```make -j6 package``` on the CMake-generated makefile in an Ubuntu 18 container:

```
/opt/irods-externals/pistache0.0.2-0/include/pistache/cookie.h:56:26: warning: 'iterator<std::bidirectional_iterator_tag, Pistache::Http::Cookie>' is deprecated [-Wdeprecated-declarations]
  struct iterator : std::iterator<std::bidirectional_iterator_tag, Cookie> {
                         ^
/opt/irods-externals/clang13.0.0-0/include/c++/v1/__iterator/iterator.h:27:29: note: 'iterator<std::bidirectional_iterator_tag, Pistache::Http::Cookie>' has been explicitly marked deprecated here
struct _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 iterator
                            ^
/opt/irods-externals/clang13.0.0-0/include/c++/v1/__config:1016:39: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
#  define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
                                      ^
/opt/irods-externals/clang13.0.0-0/include/c++/v1/__config:993:48: note: expanded from macro '_LIBCPP_DEPRECATED'
#    define _LIBCPP_DEPRECATED __attribute__ ((deprecated))
                                               ^
In file included from /irods_client_rest_cpp/irods_specific_implementation/utils.cpp:1:
In file included from /irods_client_rest_cpp/irods_specific_implementation/utils.hpp:8:
In file included from /opt/irods-externals/spdlog1.9.2-1/include/spdlog/spdlog.h:12:
In file included from /opt/irods-externals/spdlog1.9.2-1/include/spdlog/common.h:36:
/opt/irods-externals/spdlog1.9.2-1/include/spdlog/fmt/fmt.h:22:14: fatal error: 'spdlog/fmt/bundled/core.h' file not found
#    include <spdlog/fmt/bundled/core.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning and 1 error generated.
CMakeFiles/utils.dir/build.make:75: recipe for target 'CMakeFiles/utils.dir/irods_specific_implementation/utils.cpp.o' failed
make[2]: *** [CMakeFiles/utils.dir/irods_specific_implementation/utils.cpp.o] Error 1
CMakeFiles/Makefile2:156: recipe for target 'CMakeFiles/utils.dir/all' failed
make[1]: *** [CMakeFiles/utils.dir/all] Error 2
Makefile:155: recipe for target 'all' failed
make: *** [all] Error 2
```

Reply:
This has been resolved.